### PR TITLE
Dev new UI backlogs v7

### PR DESCRIPTION
**Pull request by lyazii22**
> Backlog list
- [x] DDL update on primary key update
- [x] Session listing database name updated with actual database name
- [ ] Extract interleave parent from conversion file instead of ddl
- [x] Cancel button for editing columns, primary key, foreign key and index added
- [ ] Index ordering bug fix
- [x] Move common CSS move to global style sheet
- [x] Database name extraction from session file
- [ ] Remove interleave suggestion when a table gets interleaved

### CODE REVIEW

File: ui/src/styles.scss [hunk 1]
```diff
@@ -44,6 +44,36 @@ body {
   color: #3367d6;
 }
```
Comment (shreyakhajanchi) [inline ui/src/styles.scss:46]: There are other common css that can be moved to styles.scss too, will this PR address only subset of those or is it supposed to have all the common css movement to styles.scss ?

File: ui/src/app/components/object-detail/object-detail.component.html [hunk 2]
```diff
@@ -750,13 +750,12 @@ <h3 class="title">
       <ng-container matColumnDef="spIndexOrder">
         <th class="table_header" mat-header-cell *matHeaderCellDef>Column Order</th>
         <td mat-cell *matCellDef="let element">
-          <div *ngIf="isIndexEditMode && element.get('spOrder').value">
+          <div *ngIf="isIndexEditMode && element.get('spColName').value">
```
Comment (shreyakhajanchi) [inline ui/src/app/components/object-detail/object-detail.component.html:753]: Dropping a column from index in spanner is showing removed from source as well - https://drive.google.com/file/d/1RHiW76ff1qjY88xeV28pLjR_VWAZTiaA/view?usp=sharing&resourcekey=0-nZXyKP9RKdVXwa1lCsku7g

File: ui/src/app/components/object-detail/object-detail.component.html [hunk 3]
```diff
@@ -750,13 +750,12 @@ <h3 class="title">
...
             <input
               matInput
               class="order_input"
-              type="text"
+              type="number"
```
Comment (shreyakhajanchi) [inline ui/src/app/components/object-detail/object-detail.component.html:757]: While adding a column back into primary key it takes the older order and results in issue : https://drive.google.com/file/d/1m5Rj2VVCWE0pCCsnTqq8SDITafzWcuRQ/view?usp=sharing&resourcekey=0-xCKQPLqlQ8SeZo81oTkzEQ . And there is no way to add that column at the end.

File: webv2/web.go [hunk 4]
```diff
@@ -656,6 +673,24 @@ func setParentTable(w http.ResponseWriter, r *http.Request) {
 	}
 	tableInterleaveStatus := parentTableHelper(table, update)
 
+	if tableInterleaveStatus.Possible {
```
Comment (shreyakhajanchi) [inline webv2/web.go:676]: After removing a column from primary key after interleaving the table it still shows as interleaved : https://drive.google.com/file/d/1m5Rj2VVCWE0pCCsnTqq8SDITafzWcuRQ/view?usp=sharing&resourcekey=0-xCKQPLqlQ8SeZo81oTkzEQ

**Review by shreyakhajanchi — COMMENTED** [review]
Left few comments , mostly seeing issues related to primary key/index edits

File: ui/src/app/components/object-detail/object-detail.component.html (hunk 2 above)
Comment (Ronak-Searce) [reply ui/src/app/components/object-detail/object-detail.component.html:753]: Solved.

File: webv2/web.go (hunk 4 above)
Comment (shreyakhajanchi) [reply webv2/web.go:676]: After we convert a table to interleaved and remove a column from the primary key of the interleaved table , the status still shows as interleaved.

File: ui/src/app/components/object-detail/object-detail.component.html (hunk 3 above)
[the remark this answers is not in the document]
Comment (shellymathew98) [reply ui/src/app/components/object-detail/object-detail.component.html:757]: Solved

Comment (shellymathew98) [reply ui/src/app/components/object-detail/object-detail.component.html:757]: Solved

File: webv2/web.go (hunk 4 above)
Comment (shellymathew98) [reply webv2/web.go:676]: Solved

File: ui/src/styles.scss (hunk 1 above)
Comment (lyazii22) [reply ui/src/styles.scss:46]: Consider this as a subset of common css. We will move in the comming  backlog PRs.

File: Makefile [hunk 5]
```diff
@@ -28,3 +28,11 @@ test:
...
 	go test ./... -coverprofile coverage.out -covermode count
 	go tool cover -func coverage.out
+
+ng-build:
+	cd ./ui && npm install && npm run build
+
+run-binary:
+	./harbourbridge -webv2
+
+run-all:	ng-build build run-binary
```
Comment (shreyakhajanchi) [inline Makefile:38]: If this will start the UI in a single command then please update the README accordingly. If not then what's the purpose and should it be in a separate PR ?

File: ui/src/app/components/object-detail/object-detail.component.html [hunk 6]
```diff
@@ -783,13 +783,12 @@ <h3 class="title">
...
```
Comment (shreyakhajanchi) [inline ui/src/app/components/object-detail/object-detail.component.html:790]: Weird issue with PK edit , steps to reproduce -
1. Change the order of primary key and click on save
2. Add a column to PK and click on cancel.
You will see that the UI shows the reverted order.

File: ui/src/app/components/object-detail/object-detail.component.html [hunk 7]
```diff
@@ -783,13 +783,12 @@ <h3 class="title">
       <ng-container matColumnDef="spIndexOrder">
         <th class="table_header" mat-header-cell *matHeaderCellDef>Column Order</th>
         <td mat-cell *matCellDef="let element">
-          <div *ngIf="isIndexEditMode && element.get('spOrder').value">
+          <div *ngIf="isIndexEditMode && element.get('spColName').value">
```
Comment (shreyakhajanchi) [inline ui/src/app/components/object-detail/object-detail.component.html:786]: If I try to change the order of columns in indexes it reverts to original value

File: ui/src/app/components/object-detail/object-detail.component.html [hunk 8]
```diff
@@ -852,17 +851,54 @@ <h3 class="title">
...
+      <button mat-raised-button color="primary" (click)="saveIndex()">SAVE & CONVERT</button>
+      <button
+        class="cancel-button"
+        mat-raised-button
+        color="primary"
+        (click)="toggleIndexEdit()"
+        *ngIf="currentObject!.isSpannerNode"
+      >
+        CANCEL
+      </button>
```
Comment (shreyakhajanchi) [inline ui/src/app/components/object-detail/object-detail.component.html:901]: On editing primary order based on suggestion to make the interleavable , getting the msg to go to interleave tab and convert to interleave but no interleave tab appears - https://drive.google.com/file/d/1tNmiMy3y8E1-XNi3ZHCHMCljYv6P6d_S/view?usp=sharing&resourcekey=0-3jAz7JRMQtlFqD4naku6Nw

File: ui/src/app/components/object-detail/object-detail.component.ts [hunk 9]
```diff
@@ -720,20 +782,22 @@ export class ObjectDetailComponent implements OnInit {
     })
   }
   dropIndexKey(index: number) {
-    for (let i = 0; i < this.indexData.length; i++) {
-      if (i === index || this.indexData[i].spColName === undefined) {
-        this.indexData.splice(index, 1)
-      }
+    if (this.localIndexData[index].spColName) {
+      this.localIndexData[index].spColName = undefined
```
Comment (shreyakhajanchi) [inline ui/src/app/components/object-detail/object-detail.component.ts:786]: rather than setting it to undefined is there no way to delete the property ?

File: webv2/primarykey/primarykey_handler.go [hunk 10]
```diff
@@ -118,6 +118,7 @@ func PrimaryKey(w http.ResponseWriter, r *http.Request) {
 		}
 	}
 
+	RemoveInterleave(sessionState.Conv, spannerTable)
```
Comment (shreyakhajanchi) [inline webv2/primarykey/primarykey_handler.go:121]: Not seeing the interleave tab is the branch not synced ?

File: Makefile (hunk 5 above)
Comment (manitgupta) [reply Makefile:38]: `run-binary` target doesn't make sense. The user should be able to run the webUI without having access to the source code and running a `make` target.

The built `harbourbridge` binary should directly be used, just like for all other commands today `data`, `schema` etc. This binary is built today via `build` and `build-static` targets. Please modify them. 

As per Shreya's suggestion, I would also recommend to put the `Makefile` relevant changes into a separate PR.

Comment (shellymathew98) [reply Makefile:38]: Sure Manit. Will discuss about this with team and make it a separate PR.

File: ui/src/app/components/object-detail/object-detail.component.html (hunk 8 above)
Comment (shellymathew98) [reply ui/src/app/components/object-detail/object-detail.component.html:901]: Can you take the latest pull and test if possible. Interleave tab is appearing for me.

File: ui/src/app/components/object-detail/object-detail.component.html (hunk 6 above)
Comment (shellymathew98) [reply ui/src/app/components/object-detail/object-detail.component.html:790]: Solved

File: ui/src/app/components/object-detail/object-detail.component.html (hunk 7 above)
Comment (shellymathew98) [reply ui/src/app/components/object-detail/object-detail.component.html:786]: Solved

File: ui/src/app/components/object-detail/object-detail.component.ts (hunk 9 above)
Comment (shellymathew98) [reply ui/src/app/components/object-detail/object-detail.component.ts]: Set to '' for now. If deleted entirely Source column in index will also go. Some changes will be coming here after MS5 gets merged. There we have altered the datasources for tables a little bit.

File: ui/src/app/components/object-detail/object-detail.component.html (hunk 8 above)
Comment (shreyakhajanchi) [reply ui/src/app/components/object-detail/object-detail.component.html:901]: showing now, it was due to some issue related to primary key order edit i guess

**Review by shreyakhajanchi — COMMENTED** [review]
LGTM overall except for one use case that isn't working as expected.

File: ui/src/app/components/object-detail/object-detail.component.html (hunk 7 above)
[the remark this answers is not in the document]
Comment (shellymathew98) [reply ui/src/app/components/object-detail/object-detail.component.html:786]: Solved.

[the remark this answers is not in the document]
Comment (shellymathew98) [reply ui/src/app/components/object-detail/object-detail.component.html:781]: Column order is not needed in the spanner side as per figma design(https://www.figma.com/file/mhSfts8ey0XxeHXyXdP1aC/HarbourBridge-UI?node-id=97%3A60). We removed that spOrder column in milestone 5 branch. Will remove that column from this branch too.

[the remark this answers is not in the document]
Comment (shreyakhajanchi) [reply ui/src/app/components/object-detail/object-detail.component.html:781]: If that's the case, any idea why was it added it in the first place?

File: ui/src/app/components/object-detail/object-detail.component.html [hunk 11]
```diff
@@ -854,17 +846,54 @@ <h3 class="title">
...
```
Comment (shreyakhajanchi) [inline ui/src/app/components/object-detail/object-detail.component.html:896]: After interleaving a table then editing primary key to drop the column that was used in interleaving then adding it back is not showing interleave suggestion

File: ui/src/app/components/object-detail/object-detail.component.html (hunk 7 above)
[the remark this answers is not in the document]
Comment (shellymathew98) [reply ui/src/app/components/object-detail/object-detail.component.html:781]: Solved.

[the remark this answers is not in the document]
Comment (shreyakhajanchi) [reply ui/src/app/components/object-detail/object-detail.component.html:781]: Issue doesn't seem to be solved: https://drive.google.com/file/d/1hRIzCRSeoySCud10_0ASb0o52Y_Dvsll/view?usp=sharing&resourcekey=0-bE8F_O-1qv-yYtPYAK5DBQ

**Review by shreyakhajanchi — APPROVED** [review]
LGTM
